### DISCUSSION TITLE
FancyBox HTML plugin - box sizes to correct height but goes full width

### DISCUSSION THREAD

**Discussion by Jamie-99a** [fancyapps/ui#658] [Q&A]
I'm not sure if I'm doing something wrong, but when using the HTML iframe plugin line so:

`<a href="inline.html" data-fancybox data-type="iframe">View details</a>`

the box that pops up correctly sizes its height based on the contents of "inline.html", however the width of the box is the entire width of the browser window rather than just being wide enough to encompass "inline.html".

Is there some option I am missing?

**Comment by fancyapps** [comment]
Hi,

Well, if you do not set specific dimensions, then HTML block elements will freely expand horizontally.
Either set max width for your entire document using CSS or add, for example, `data-width` attribute to your link to set max horizontal dimension.

**Comment by Jamie-99a** [comment]
Thanks for your prompt response.

Within "inline.html" (the HTML document that is to display in the box) I have tried these CSS declarations:

`html { width: 300px; }`
`body { width: 300px; }`

So everything in that document gets restricted to 300 pixels wide and text does correctly wrap around, however FancyBox still displays it full width with lots of empty space on the right.

I would prefer not to use "data-width=300" in the parent document as the page width specified by CSS within "inline.html" may change at times and I would then have to change every link to it.

**Reply by fancyapps** [reply]
Yeah, sorry about that, it will be fixed in the next minor release. But it looks like you're the only one using this feature, so it's unlikely to be included in the next major release.
